__tests__/endpoints.test.js

```diff
@@ -0,0 +1,348 @@
+                body: "I am northcoder student"
+        })
+        .expect(404)
+        .then(({body}) => {
+            expect(body).toMatchObject({ status: 404, msg: 'Not found' })
+        });
+    })
+
+    test('Status 401: unauthorised user tried to add a comment to an article',()=>{
+        return request(app)
+        .post('/api/articles/3/comments')
+        .send({ username: "daniel",
+                body: "I am northcoder student"
+        })
+        .expect(401)
+        .then(({body}) => {
+            expect(body).toMatchObject({ status: 401, msg: 'Unauthorized' })
+        });
+    })
+
+
+    
+})
+
+
+
+describe('Patches the article by updating the number of votes', () => {
+    test('Status 201: succesfully patches the article when a valid article is chosen and updates the number of votes when given amount to update them',()=>{
+        return request(app)
+        .patch('/api/articles/3')
+        .send({ inc_votes: 50 })
+        .expect(201)
+        .then(({body}) => {
+            expect(body).toMatchObject({
+                article_id: 3,
+                title: 'Eight pug gifs that remind me of mitch',
+                topic: 'mitch',
+                author: 'icellusedkars',
+                body: 'some gifs',
+                created_at: '2020-11-03T09:12:00.000Z',
+                votes: 50,
+                article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'})
+        });
+    })
+
+    test('Status 404: tries to patch a non existant article and returns an error and status code',()=>{
+        return request(app)
+        .patch('/api/articles/999')
+        .send({ inc_votes: 50 })
+        .expect(404)
+        .then(({body}) => {
+            expect(body).toMatchObject({ status: 404, msg: 'Not found' })
+        })
+    })
+  
+
+})
+
+describe('allows user to filter the articles by a topic query', () => {
+    test('Status 200: returns all the articles which have a topic of cats',()=>{
+        return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
+        .then(({body}) => {
+            expect(body).toEqual( [
+                {
+                  article_id: 5,
+                  title: 'UNCOVERED: catspiracy to bring down democracy',
+                  topic: 'cats',
+                  author: 'rogersop',
+                  created_at: '2020-08-03T13:14:00.000Z',
+                  votes: 0,
+                  article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+                }
+              ])
+        })
+        
+    })
+
+    test('Status 404: returns 404 when no such topic in the articles',()=>{
+        return request(app)
+        .get('/api/articles?topic=elephants')
+        .expect(404)
+        .then(({body}) => {
+           expect(body).toMatchObject({ status: 404, msg: 'Not Found' })
+        })
+        
+    })
```

Comment:
It's worth testing what happens with an existent topic query, but has no articles responds with an empty array of articles, e.g. ?topic=paper. (expect 200, [])